pkg/proto/interface.go

```diff
@@ -72,6 +72,12 @@ type (
 		PostHandle(ctx context.Context, result Result) error
 	}
 
+	DBConnectionMetricFilter interface {
+		Filter
+		PreHandle(ctx context.Context, conn Connection) error
+		PostHandle(ctx context.Context, result Result, conn Connection) error
+	}
+
```

Comment:
this interface is unnecessary! metricsFilter can be asserted as DBConnectionPreFilter and DBConnectionPostFilter